src/test/java/com/williamfiset/algorithms/datastructures/stack/StackTest.java

```diff
@@ -12,7 +13,8 @@
 
   @Before
   public void setup() {
-    stack = new Stack<Integer>();
+    stack = new ListStack<Integer>();
```

Comment:
Let's test both the ListStack and the ArrayStack. Can you update the tests to be more generic? For example, maintain a list of java.util.Stack interface implementations and use that in each test?

for instance, something like

List<Stack<Integer>> stacks;

public void setup() {
stacks.add(new ListStack<Integer>());
stacks.add(new ArrayStack<Integer>());
}

in each test loop over all stack impls:

public void sometest() {
  for (Stack stack : stacks) {
     ... // test 'stack'
  }
}

Reply:
has fixed